### DISCUSSION TITLE
Why do `dt.sum()` and `dt.mean()` act differently, when using multiple columns?

### DISCUSSION THREAD

**Discussion by lmullany** [h2oai/datatable#3405] [Q&A]
If I use the example `Frame` in the documentation https://datatable.readthedocs.io/en/v1.0.0/manual/comparison_with_rdatatable.html as below:

```python
from datatable import dt, f, g, by, update, join, sort

DT = dt.Frame(x = ["b"]*3 + ["a"]*3 + ["c"]*3,
              y = [1, 3, 6] * 3,
              v = range(1, 10))
```

Using `dt.mean()`, the mean of each **column** is returned as follows
```python
DT[:, dt.mean([f.y,f.v])]

   |       y        v
   | float64  float64
-- + -------  -------
 0 | 3.33333        5
[1 row x 2 columns]
```

However, the exact same syntax, using `dt.sum()` instead of `dt.mean()`, works **row-wise**
```python 
DT[:, dt.sum([f.y,f.v])]

   |    C0
   | int32
-- + -----
 0 |     2
 1 |     5
 2 |     9
 3 |     5
 4 |     8
 5 |    12
 6 |     8
 7 |    11
 8 |    15
[9 rows x 1 column]
```

Why is this happening?

**Comment by oleksiyskononenko** [comment]
I think the reason for this is that we're trying not to shadow the built-in python `sum()` with the following wrapper:
https://github.com/h2oai/datatable/blob/main/src/datatable/expr/reduce.py#L111-L115

This wrapper is looking at the type of the argument, and for `FExpr` it invokes datatable's `sum()`, while for all the other types it is the built-in python's `sum()` that is invoked.

In your example, the argument is not an `FExpr`, but a list of `FExpr`s, for which we wrongly call the built-in python `sum()`. 

It seems that we need to review how this wrapper is implemented. For now, to make it work, you can simply do 
```python
DT[:, [dt.sum(f.y), dt.sum(f.v)]]
```

**Reply by oleksiyskononenko** [reply]
Fixed as of https://github.com/h2oai/datatable/pull/3407
